src/server.c

```diff
@@ -186,9 +188,11 @@ int dqlite__init(struct dqlite_node *d,
 err_after_raft_transport_init:
 	raftProxyClose(&d->raft_transport);
 err_after_pool_init:
+#ifdef DQLITE_NEXT
 	pool_close(&d->pool);
 	pool_fini(&d->pool);
 err_after_loop_init:
```

Comment:
I may be wrong here but I think the label that exists in `master` at the moment is `err_after_loop_init` so the if def should be:
```c
....
err_after_raft_transport_init:
	raftProxyClose(&d->raft_transport);
#ifdef DQLITE_NEXT
err_after_pool_init:
	pool_close(&d->pool);
	pool_fini(&d->pool);
#endif
err_after_loop_init:
	uv_loop_close(&d->loop);
....
```

Reply:
That way of doing it doesn't quite work as-is because there is a `goto err_after_pool_init` in the error branch after `raftProxyInit` above. We would have to change that to

```
        rv = raftProxyInit(&d->raft_transport, &d->loop);
        if (rv != 0) {
#ifdef DQLITE_NEXT
                goto err_after_pool_init;
#else
                goto err_after_loop_init;
#endif
        }
```

The way I've written it looks kind of weird but gets away with only one `ifdef`, which I think is nice. Thoughts?